moban/plugins.py

```diff
@@ -89,7 +97,15 @@ def _render_with_finding_template_first(self, template_file_index):
                 self.template_dirs, template_file
             )
             for (data_file, output) in data_output_pairs:
-                data = self.context.get_data(data_file)
+                try:
```

Comment:
Logically, there is nothing wrong with this implementation. I am thinking of one alternative, which is to inject environment variables to jinja2 globals, which then automatically brings them into all templates. 

One drawback is: what happens when user defined data field conflicts with environment variable?

Reply:
@jayvdb what's your view since you had this original ticket?

Reply:
The env vars should take precedence over data from config files.

There should have a test case which demonstrates this.

However it would be good to emit a notice whenever an environment variable overrides a static config data value.  We might defer that to a separate patch, as it might get a bit complicated because we dont want a notice for environment variables which have the same value as a config data value.

This PR needs new test cases of the added functionality before it adds more logic to the startup process.

Reply:
Alright! @CLiu13 , can I put one more challenge to you? As you can see, there are 3 places having similar codes. With DIY principle, could you please refactor the code?